Mathlib/Algebra/Group/Pointwise/Finset/Basic.lean

```diff
@@ -946,29 +949,70 @@ protected def monoid : Monoid (Finset α) :=
 
 scoped[Pointwise] attribute [instance] Finset.monoid Finset.addMonoid
 
+-- `Finset.pow_left_mono` doesn't exist since it would syntactically be a special case of
+-- `pow_left_mono`
+
 @[to_additive]
-theorem pow_mem_pow (ha : a ∈ s) : ∀ n : ℕ, a ^ n ∈ s ^ n
-  | 0 => by
-    simp only [pow_zero, mem_one]
-  | n + 1 => by
-    simp only [pow_succ]
-    exact mul_mem_mul (pow_mem_pow ha n) ha
+protected lemma pow_right_mono (hs : 1 ∈ s) : Monotone (s ^ ·) :=
+  pow_right_monotone <| one_subset.2 hs
+
+@[to_additive (attr := gcongr)]
+lemma pow_subset_pow_left (hst : s ⊆ t) : s ^ n ⊆ t ^ n := pow_left_mono _ hst
+
+@[to_additive (attr := gcongr)]
+lemma pow_subset_pow_right (hs : 1 ∈ s) (hmn : m ≤ n) : s ^ m ⊆ s ^ n :=
+  Finset.pow_right_mono hs hmn
+
+@[to_additive (attr := gcongr)]
+lemma pow_subset_pow (hst : s ⊆ t) (ht : 1 ∈ t) (hmn : m ≤ n) : s ^ m ⊆ t ^ n :=
+  (pow_subset_pow_left hst).trans (pow_subset_pow_right ht hmn)
+
+@[deprecated (since := "2024-11-19")] alias pow_subset_pow_of_one_mem := pow_subset_pow_right
+
+@[deprecated (since := "2024-11-19")]
+alias nsmul_subset_nsmul_of_zero_mem := nsmul_subset_nsmul_right
 
 @[to_additive]
-theorem pow_subset_pow (hst : s ⊆ t) : ∀ n : ℕ, s ^ n ⊆ t ^ n
-  | 0 => by
-    simp [pow_zero]
-  | n + 1 => by
-    rw [pow_succ]
-    exact mul_subset_mul (pow_subset_pow hst n) hst
+lemma pow_subset_pow_mul_of_sq_subset_mul (hst : s ^ 2 ⊆ t * s) (hn : 1 < n) :
+    s ^ n ⊆ t ^ (n - 1) * s := pow_le_pow_mul_of_sq_le_mul hst hn
+
+@[to_additive (attr := simp) nsmul_empty]
+theorem empty_pow (hn : n ≠ 0) : (∅ : Finset α) ^ n = ∅ := by
+  rw [← tsub_add_cancel_of_le (Nat.succ_le_of_lt <| Nat.pos_of_ne_zero hn), pow_succ', empty_mul]
```

Comment:
```suggestion
theorem empty_pow (hn : n ≠ 0) : (∅ : Finset α) ^ n = ∅ :=
  match n with | n + 1 => by simp [pow_succ]
```
imo this is quite a bit cleaner. 
(aside, there's also the slightly funny proof
```
simpa using pow_subset_pow_mul_of_sq_subset_mul (α := α) (pow_two ∅).le hn
```
)

Reply:
Wow yes that's a funny proof